### ISSUE TITLE
File uploads

### ISSUE THREAD

**Issue by autowp** [appleboy/gofight#55]
Missing the way to test `multipart/form-data` encoded forms and file uploads

**Comment by jcoyne** [comment]
Here's how I worked around:
```
	r.POST("/v1/file").
		SetHeader(gofight.H{
			"Content-Type": "multipart/form-data; boundary=------------------------a31e2ddd4b2c0d92",
		}).
		SetBody(`--------------------------a31e2ddd4b2c0d92
Content-Disposition: form-data; name="upFile"; filename="foo.txt"
Content-Type: text/plain

Hello

--------------------------a31e2ddd4b2c0d92--`).
Run(...)
```

**Comment by appleboy** [comment]
See PR #66 

route:

```go
func gintFileUploadHandler(c *gin.Context) {
	file, err := c.FormFile("test")
	if err != nil {
		c.JSON(http.StatusBadRequest, gin.H{
			"error": err.Error(),
		})
		return
	}
	foo := c.PostForm("foo")
	bar := c.PostForm("bar")
	c.JSON(http.StatusOK, gin.H{
		"hello":    "world",
		"filename": file.Filename,
		"foo":      foo,
		"bar":      bar,
	})
}
```

test code:

```go
func TestUploadFile(t *testing.T) {
	r := New()

	r.POST("/upload").
		SetFileFromPath("fixtures/hello.txt", "test", H{
			"foo": "bar",
			"bar": "foo",
		}).
		Run(framework.GinEngine(), func(r HTTPResponse, rq HTTPRequest) {
			data := []byte(r.Body.String())

			hello := gjson.GetBytes(data, "hello")
			filename := gjson.GetBytes(data, "filename")
			foo := gjson.GetBytes(data, "foo")
			bar := gjson.GetBytes(data, "bar")

			assert.Equal(t, "world", hello.String())
			assert.Equal(t, "hello.txt", filename.String())
			assert.Equal(t, "bar", foo.String())
			assert.Equal(t, "foo", bar.String())
			assert.Equal(t, http.StatusOK, r.Code)
			assert.Equal(t, "application/json; charset=utf-8", r.HeaderMap.Get("Content-Type"))
		})
}func TestUploadFile(t *testing.T) {
	r := New()

	r.POST("/upload").
		SetFileFromPath("fixtures/hello.txt", "test", H{
			"foo": "bar",
			"bar": "foo",
		}).
		Run(framework.GinEngine(), func(r HTTPResponse, rq HTTPRequest) {
			data := []byte(r.Body.String())

			hello := gjson.GetBytes(data, "hello")
			filename := gjson.GetBytes(data, "filename")
			foo := gjson.GetBytes(data, "foo")
			bar := gjson.GetBytes(data, "bar")

			assert.Equal(t, "world", hello.String())
			assert.Equal(t, "hello.txt", filename.String())
			assert.Equal(t, "bar", foo.String())
			assert.Equal(t, "foo", bar.String())
			assert.Equal(t, http.StatusOK, r.Code)
			assert.Equal(t, "application/json; charset=utf-8", r.HeaderMap.Get("Content-Type"))
		})
}
```

**Comment by appleboy** [comment]
@autowp @jcoyne @lvxuan263 I already merged #66 PR. Please try the latest code. I will update the document.
